### PR TITLE
DAOS-3212 mgmt: add log message API

### PR DESCRIPTION
**Pull request by wangdi1**
Add log message API to set mark to server log
for debugging purpose.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

File: src/mgmt/srv.c [hunk 1]
```diff
@@ -1099,6 +1099,54 @@ ds_mgmt_profile_hdlr(crt_rpc_t *rpc)
...
+	int			rc;
+
+	in = crt_req_get(rpc);
+	D_ASSERT(in != NULL);
+
+	topo = crt_tree_topo(CRT_TREE_KNOMIAL, 32);
+	opc = DAOS_RPC_OPCODE(MGMT_TGT_MARK, DAOS_MGMT_MODULE,
+			      DAOS_MGMT_VERSION);
+	rc = crt_corpc_req_create(dss_get_module_info()->dmi_ctx, NULL, NULL,
+				  opc, NULL, NULL, 0, topo, &tc_req);
```
Comment (liw) [inline src/mgmt/srv.c]: You guys are aware that using this in-band approach we will suffer from killed ranks, right?

Comment (wangdi1) [reply src/mgmt/srv.c]: yes, but this is the easy way to ship the function at the moment. And given the use case, is probably ok for now. though we probably can retrieve the killed ranks from management server and skip them in future.

File: src/mgmt/srv_target.c [hunk 2]
```diff
@@ -795,6 +795,25 @@ ds_mgmt_tgt_profile_hdlr(crt_rpc_t *rpc)
...
+void
+ds_mgmt_tgt_mark_hdlr(crt_rpc_t *rpc)
+{
+	struct mgmt_mark_in	*in;
+	struct mgmt_mark_out	*out;
+
+	in = crt_req_get(rpc);
+	D_ASSERT(in != NULL);
+
+	D_DEBUG(DB_TRACE, "Mark trace %s.\n", in->m_mark);
```
Comment (liw) [inline src/mgmt/srv_target.c]: If this is to be used for marking test boundaries, we might want a high log level.

Comment (wangdi1) [reply src/mgmt/srv_target.c]: probably this mask should be an input parameter. but export DB_X seems not a nice thing to do? any suggestion?

Comment (liuxuezhao) [reply src/mgmt/srv_target.c]: Maybe just using D_CRIT(...), as when user call that API it want that mark msg shows there.

File: src/mgmt/cli_mgmt.c [hunk 3]
```diff
@@ -222,6 +222,45 @@ dc_mgmt_profile(uint64_t modules, char *path, bool start)
...
+	if (rc != 0) {
+		D_ERROR("crt_req_create failed, rc: %d.\n", rc);
+		D_GOTO(err_grp, rc);
+	}
+
+	D_ASSERT(rpc != NULL);
+	in = crt_req_get(rpc);
+	in->m_mark = (char *)mark;
+	/** send the request */
+	rc = daos_rpc_send_wait(rpc);
```
Comment (liuxuezhao) [inline src/mgmt/cli_mgmt.c]: I am not sure if the wait is good, as it need to wait bcast at server side.
if it is only used in testing code should be fine.

Comment (wangdi1) [reply src/mgmt/cli_mgmt.c]: Indeed, but this is what we do for synchronous API? or you mean sth else?

**Pull request merged**